solver/src/http_solver.rs

```diff
@@ -91,11 +96,27 @@ impl HttpSolver {
             .collect()
     }
 
-    fn token_models(&self, tokens: &HashMap<String, H160>) -> HashMap<String, TokenInfoModel> {
-        tokens
+    async fn token_models(
+        &self,
+        tokens: &HashMap<String, H160>,
+    ) -> Result<HashMap<String, TokenInfoModel>> {
+        let addresses: Vec<H160> = tokens.values().into_iter().cloned().collect();
+        let token_infos = self
+            .token_info_fetcher
+            .get_token_infos(addresses.as_slice())
+            .await?;
```

Comment:
Wanted to ask this again: do we really want to replace missing token info by some default (e.g. 0 decimals) and try the solver anyway instead?

Reply:
I'm fine with either approach. If we skip orders with tokens without decimals then the solver can't handle them at all. Is there a disadvantage to using a default value? Does it make the solver worse at the other tokens if one of them has very wrong decimals?

Reply:
>  Does it make the solver worse at the other tokens if one of them has very wrong decimals?

That may very well happen yes. We are currently relying on number of decimals for avoiding numerical issues in the solver, so having a wrong number for one token can be enough to invalidate the full solution.

Reply:
I guess a more reliable solution would be to make number of decimals optional for the solver, but given that we're building a PoC and this is not supposed to be common, I'd say to just skip the MIP solver whenever there are orders involving tokens without decimals for now. @twalth3r  What is your opinion?

Reply:
I would filter out the orders for which we don't have decimals and continue with the rest.

Reply:
I'd also be fine with both, hoping it wouldn't happen too often anyways.

If we're going for the default-approach, maybe using a default of 18 decimals is safer here!? Most tokens have 18 decimals, so chances are good we're on the right path. If the token actually has fewer decimals, then the order amounts should be accordingly small, and our solver would just treat them as negligible under its 18-decimal assumption. Does that make sense?

Reply:
I think I'd also slightly prefer filtering out non-compliant tokens. We can add support once we see a real use case and this way we don't risk unexpected solver behavior.

Reply:
This was changed now: driver passes a list with all tokens and all orders to solver, and the solver does its best with it, which may fallback to ignoring the tokens without decimals.